docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+supervisor for managing the containers and the [workload](#workload)
+running within those containers.
+
+> **Note:**
+>
+> Only a single agent process is run for each VM created.
+
+### Agent communications protocol
+
+The agent communicates with the other Kata components using a
+[`ttRPC`](https://github.com/containerd/ttrpc-rust) based
+[protocol](../../src/agent/protocols/protos).
+
+See also the [agent control tool](#agent-control-tool).
+
+## Workload
+
+The workload is the OCI bundle rootfs. It is exported from the host
+into the VM.
 
-For the 2.0 release, the `kata-agent` is rewritten in the [RUST programming language](https://www.rust-lang.org/) so that we can minimize its memory footprint while keeping the memory safety of the original GO version of [`kata-agent` used in Kata Container 1.x](https://github.com/kata-containers/agent). This memory footprint reduction is pretty impressive, from tens of megabytes down to less than 100 kilobytes, enabling Kata Containers in more use cases like functional computing and edge computing.
+### Workload root filesystem
 
-The `kata-agent` execution unit is the sandbox. A `kata-agent` sandbox is a container sandbox defined by a set of namespaces (NS, UTS, IPC and PID). `shimv2` can
-run several containers per VM to support container engines that require multiple
-containers running inside a pod.
+For details of how the [runtime](#runtime) make the workload image
+chosen by the user (such as `busybox:latest`) available to the
+workload process, see the [storage](#storage) section.
 
-`kata-agent` communicates with the other Kata components over `ttRPC`.
+The container workload process defined by the OCI bundle's
+configuration file (`config.json`) is created using the Kata
+Containers [agent](#agent). The agent creates a container environment
+to run the workload in, in a similar manner to that done by the
+[`runc`](https://github.com/opencontainers/runc) OCI runtime, using
+Linux cgroups and namespaces.
+
+Note that the workload is isolated from the VM root environment by its
+surrounding container environment. The VM environment the workload
+container runs in is also isolated from the "outer" host environment
+where the container manager runs.
 
 ## Runtime
 
-`containerd-shim-kata-v2` is a [containerd runtime shimv2](https://github.com/containerd/containerd/blob/v1.4.1/runtime/v2/README.md) implementation and is responsible for handling the `runtime v2 shim APIs`, which is similar to [the OCI runtime specification](https://github.com/opencontainers/runtime-spec) but simplifies the architecture by loading the runtime once and making RPC calls to handle the various container lifecycle commands. This refinement is an improvement on the OCI specification which requires the container manager call the runtime binary multiple times, at least once for each lifecycle command.
+The Kata Containers runtime (or shim), is a [shimv2](#shim-v2-architecture) compatible runtime.
```

Comment:
I think kata runtime, runtime, shim, are being used as the same thing for way too long to just get an explanation about it now.

Would it make sense to, maybe, clarify that those are all the same earlier in the document and, maybe, refer to it always as "shim"?